WeaselTSF/CandidateList.cpp

```diff
@@ -423,7 +424,33 @@ void WeaselTSF::_SelectCandidateOnCurrentPage(size_t index)
 	m_client.SelectCandidateOnCurrentPage(index);
 	// fake a emptyp presskey to get data back and DoEditSession
 	m_client.ProcessKeyEvent(0);
-	_UpdateComposition(_pEditSessionContext);
+	{
+		std::wstring commit;
+		weasel::Config config;
+		auto context = std::make_shared<weasel::Context>();
+		weasel::ResponseParser parser(&commit, context.get(), &_status, &config, &_cand->style());
+		bool ok = m_client.GetResponseData(std::ref(parser));
+		_UpdateLanguageBar(_status);
+		if (ok)
+		{
+			if (!commit.empty())
+			{
+				_InsertText(_pEditSessionContext, commit);
+				_EndComposition(_pEditSessionContext, false);
+				// prediction situation, simuulate a Page_Up, to avoid crash
+				// fixme: are there better ways?
+				if (context->cinfo.candies.size()){
+					INPUT inputs[2];
+					inputs[0].type = INPUT_KEYBOARD;
+					inputs[0].ki = {VK_PRIOR, 0,0,0,0};
+					inputs[1].type = INPUT_KEYBOARD;
+					inputs[1].ki = {VK_PRIOR, 0,KEYEVENTF_KEYUP,0,0};
+					::SendInput(sizeof(inputs) / sizeof(INPUT), inputs, sizeof(INPUT));
```

Comment:
I guess we can try to defer the `_UpdateComposition` call, or send an input key right here.

Sending an input key to the system essentially postpones the key event handling.

Reply:
if only  do `_UpdateComposition` here, crash is fixed, however the prediction now shown until next key press comes.

Reply:
How did the crash happen?

How is it fixed then?

Reply:
> How did the crash happen?
> 
> How is it fixed then?

maybe some object is referenced after released, after prediction is enabled and clicking select on non-prediction candidate

maybe some status different from space-selecting or auto-committing.